src/dd_init.F90

```diff
@@ -0,0 +1,325 @@
+module dd_init
+
+  use constants
+  use dd_header,  only: dd_type
+  use error,      only: fatal_error, warning
+  use global,     only: n_procs, n_particles, rank, mpi_err, master
+  use mesh,       only: bin_to_mesh_indices, mesh_indices_to_bin
+  use output,     only: write_message
+  use search,     only: binary_search
+  use string,     only: to_str
+
+#ifdef MPI
+  use mpi
+#endif
+
+#ifdef _OPENMP
+    use omp_lib
+#endif
+
+  implicit none
+  public
+
+contains
+
+!===============================================================================
+! INITIALIZE_DOMAIN_DECOMP
+!===============================================================================
+
+  subroutine initialize_domain_decomp(dd)
+
+    type(dd_type), intent(inout) :: dd
+
+    integer :: d                ! neighbor bin
+    integer :: neighbor_meshbin
+    integer :: alloc_err        ! allocation error code
+    integer :: world_group
+    integer :: domain_master_group
```

Comment:
declarations of `world_group` and `domain_master_group` should be surrounded with `#ifdef MPI`